Modules/@babylonjs/react-native/shared/BabylonNative.cpp

```diff
@@ -142,7 +144,66 @@ namespace Babylon
                 }
             });
         }
+       
+        struct JSGuard
+        {
+            JSGuard(Dispatcher& dispatcher)
+            {
+                std::promise<void> promise;
+                std::future<void> future = promise.get_future();
+                std::future<void> blocking = get().get_future();
+                dispatcher([&]() mutable
+                {
+                    promise.set_value();
+                    blocking.wait();
+				});
+                future.wait();
+            }
+
+            ~JSGuard()
+            {
+                get().set_value();
+            }
+
+            static std::promise<void>& get()
+            {
+                static std::promise<void> promise;
+                return promise;
+            }
+        };
+
+        void EnableView()
+        {
+            if (!m_graphics)
+                return;
+            
+            if (m_autoRender)
+            {
+                m_graphics->EnableRendering();
+            }
+            else
+            {
+                JSGuard guard(m_jsDispatcher);
+                m_graphics->EnableRendering();
```

Comment:
So basically this is going to prevent the JS thread from doing anything during a call to `EnableRendering`? I think this won't be necessary after @bghgary's in flight changes that remove auto rendering and makes graphics threading much more robust. We'll probably need to wait for that change to wrap up and then see how this part of the change would apply (if at all).

Reply:
Blocks JavaScript threads of Babylon js mapped to bgfx.

I don't actually use it. I wasted time safely shutting down and recreating bgfx. Even if I shut it down safely, it crashes by bgfx related call mapped with babylonjs before bgfx is created again (so it doesn't shut down bgfx).

Reply:
thanks for the context @nkari82!